src/rustup-cli/help.rs

```diff
@@ -78,7 +78,23 @@ inferred, so the above could be written:
     $ rustup default stable-msvc
 
 Toolchain names that don't name a channel instead can be used to name
-custom toolchains with the `rustup toolchain link` command.";
+custom toolchains with the `rustup toolchain link` command. This can
+also be used to symlink toolchains from local builds. For more
+information see `rustup toolchain help link`.";
+
+pub static TOOLCHAIN_LINK_HELP: &'static str =
+r"
+Symlinks a toolchain from a local directory specified by 'path' and
+gives it a custom name specified by 'toolchain' if it does not name a
+standard release channel.
```

Comment:
(Okay, I don't fully understand _all_ the code but here is what I think is correct.)
What it means is that if you give it a name such as a standard release channel, say `nightly` then it will parse and understand what you are trying to say (https://github.com/rust-lang-nursery/rustup.rs/blob/master/src/rustup/config.rs#L388). 
You can certainly use `nightly` but that would mean that your actual `nightly` compiler (if you have one) will be replaced by the one at `path`.
If you give it some unknown name which it doesn't recognize, say `nightly1`, then it will think that you are trying to give a custom name to the toolchain and will oblige.